pkg/assembler/backends/inmem/pkg.go

```diff
@@ -98,15 +100,15 @@ type pkgVersionStruct struct {
 	id                  uint32
 	parent              uint32
 	name                string
-	versions            pkgVersionList
+	versions            pkgVersionMap
 	srcMapLinks         []uint32
 	isDependencyLinks   []uint32
 	badLinks            []uint32
 	goodLinks           []uint32
 	hasMetadataLinks    []uint32
 	pointOfContactLinks []uint32
 }
-type pkgVersionList []*pkgVersionNode
+type pkgVersionMap map[string]*pkgVersionNode
```

Comment:
to make things a bit more clearer, could define a type for `pkgVersionNodeHash` and then have `hashPkgVersionNode` return that type